cirrus/google_cloud/manager.py

```diff
@@ -334,29 +346,98 @@ def get_project_info(self):
 
         return response.json()
 
-    def get_buckets(self):
+    def get_bucket_iam_policy(self, bucket_name):
+        bucket = self._storage_client.get_bucket(bucket_name)
+        return bucket.get_iam_policy()
+
+    def create_bucket(
+            self, name, storage_class=None, public=False, requester_pays=False,
+            project=None, access_logs_bucket=None):
         """
-        Return all the buckets for the project
+        Create a Google Storage bucket.
 
         Returns:
-            List(google.cloud.storage.bucket.Bucket): Google Cloud Buckets
+            google.cloud.storage.bucket.Bucket: Google Cloud Bucket
+
+        Args:
+            name (str): Globally unique name for new Google Bucket
+            storage_class (str, optional): one of GOOGLE_STORAGE_CLASSES
+            public (bool, optional): Whether or not all data in bucket should
+                be open to the public (will only allow access by authN'd users
+                to support access logs)
+            requester_pays (bool, optional): Whether requester pays for API
+                requests for this bucket and its blobs.
+            project (str, optional): FIXME not currently used since Google's
+                Python API won't accept it... their docs are out of date and
+                new versions of the code default to client.project, which should
+                presumably be fine.
+            access_logs_bucket (str, optional): Google bucket name to store
+                access logs for this newly created bucket
+
+        Raises:
+            GoogleAuthError: Description
+            ValueError: Description
         """
         if not self._authed_session:
             raise GoogleAuthError()
 
-        buckets = list(self._storage_client.list_buckets())
-        return buckets
+        if storage_class and storage_class not in GOOGLE_STORAGE_CLASSES:
+            raise ValueError(
+                'storage_class {} not one of {}. Did not create bucket...'
+                .format(storage_class, GOOGLE_STORAGE_CLASSES))
+
+        bucket = storage.bucket.Bucket(
+            client=self._storage_client, name=name)
 
-    def create_bucket(self, name):
+        if requester_pays is not None:
+            bucket.requester_pays = requester_pays
+
+        if storage_class:
+            bucket.storage_class = storage_class
+
+        bucket.create()
+
+        if public:
+            # update bucket iam policy with allAuthN users having read access
+            policy = bucket.get_iam_policy()
+            role = GooglePolicyRole('roles/storage.objectViewer')
+            policy[str(role)] = ['allAuthenticatedUsers']
+            bucket.set_iam_policy(policy)
+
+        if access_logs_bucket:
+            bucket.enable_logging(access_logs_bucket)
+
+        bucket.update()
+
+    def give_group_access_to_bucket(self, group_email, bucket_name):
```

Comment:
can we add a parameter for access=read/write ?
we will need a role for write access for data replication service